src/model/related-article.ts

```diff
@@ -0,0 +1,40 @@
+import { BackmatterEntity } from './backmatter-entity';
+import {JSONObject} from "./types";
+
+export class RelatedArticle extends BackmatterEntity {
+  public articleType: string = '';
+  public href: string = '';
+
+  constructor(data?: JSONObject | Element) {
+    super();
+    this.createEntity(data);
+  }
+
+  public clone(): RelatedArticle {
+    const newArticle = new RelatedArticle();
+    newArticle._id = this._id;
+    newArticle.articleType = this.articleType;
+    newArticle.href = this.href;
+    return newArticle;
+  }
+
+  protected fromXML(xmlNode: Element): void {
+    this.articleType = xmlNode.getAttribute('related-article-type') || '';
+    this.href = xmlNode.getAttribute('xlink:href') || '';
+  }
+
+  protected fromJSON(json: JSONObject): void {
+    this._id = (json._id as string) || this.id;
+    this.articleType = json.articleType as string;
+    this.href = json.href as string;
+  }
+
+  protected createBlank(): void {
+    this.articleType = '';
+    this.href = '';
+  }
+}
+
+export function createRelatedArticleState(relatedArticlesXml: Element[]): RelatedArticle[] {
```

Comment:
Can we get some tests for this ensuring things like the array length is expected, the contents of array returned is expected given the parameter passed